### PR TITLE
chore: readme changes

### PR DESCRIPTION
**Pull request by irohdw**
## Description
docs: added new sections to README

Added nvm as dependency with instruction that users must install it manually.

Added information on what the .env file is, the need to get proper values for REACT_APP_PORTIS_DAPP_ID and REACT_APP_ETHEREUM_NODE_URL, what happens if they aren't added, and how to obtain them. 

Added "Getting Started" section that explains git services developers should install (Pull and Git Credential Manager) to make their work easier.

Added developer workflow section that explains the process of creating fork, how to make local changes, and how to make a pull request.
 
Added link to Lib/Contributing.md so developers understand how to make proper commits.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [X] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [X] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [X] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,33 +32,78 @@ ShapeShift's OSS 2nd generation Web application. (Under Development)
...
+
+- (optional) Make sure you're using the right Node.js version.
 
   ```sh
-  cp sample.env .env
+  nvm use
   ```
+The .env file contains global variables that the program needs to function properly. The following variables are not publicly available for security reasons:
+
+REACT_APP_PORTIS_DAPP_ID<br>
```
Comment (0xdef1cafe) [inline README.md:54]:
```suggestion
`REACT_APP_PORTIS_DAPP_ID`
```

File: README.md [hunk 2]
```diff
@@ -32,33 +32,78 @@ ShapeShift's OSS 2nd generation Web application. (Under Development)
...
+Allows you to connect a Portis wallet. Without this the program will hang after choosing Portis and 
+clicking the "Pair" button. To get it, make a post asking for it in the Discord.
+
+REACT_APP_ETHEREUM_NODE_URL <br>
```
Comment (0xdef1cafe) [inline README.md:58]:
```suggestion
`REACT_APP_ETHEREUM_NODE_URL`
```

File: README.md [hunk 3]
```diff
@@ -32,33 +32,78 @@ ShapeShift's OSS 2nd generation Web application. (Under Development)
...
+Program crashes when connecting wallet without this. Obtain by doing the following: 
+1. Go to https://infura.io/dashboard
+2. Set up a free account
+3. Make a new project 
+<br>Your key should use "JSON-RPC over HTTPS" and look like this: 
+https://mainnet.infura.io/v3/YOUR-PROJECT-ID
```
Comment (0xdef1cafe) [inline README.md:64]:
```suggestion

Your key should use "JSON-RPC over HTTPS" and look like this

`https://mainnet.infura.io/v3/YOUR-PROJECT-ID`
```

File: README.md [hunk 4]
```diff
@@ -114,3 +159,94 @@ From your projects folder:
...
+
+        (Note that the Linux packages for versions 2.0.567 and 2.0.605, which are the two latest versions at time of writing, are broken. Until a fixed version is released, you'll want to use [2.0.498](https://github.com/GitCredentialManager/git-credential-manager/releases/tag/v2.0.498).)
+
+        On Debian-based Linux distros, including Ubuntu, Mint, and Pop OS, download the `.deb` package for the version you want to install. Then run:
+
+        ```sh
+        sudo dpkg -i '<path to .deb file>'
+        git-credential-manager-core configure
+        git config --global credential.credentialStore secretservice
+        ```
```
Comment (0xdef1cafe) [inline README.md:178]: i think this is general developer skills/preferences and not specific to this repo, and should be excluded

File: README.md [hunk 5]
```diff
@@ -32,33 +32,78 @@ ShapeShift's OSS 2nd generation Web application. (Under Development)
...
+
+  Needed for certain Defi integrations such as Yearn; the app will malfunction when connecting a wallet without it.
+
+  Any Ethereum node should do, but you can get your own node URL for testing by doing the following:
+
+  1. Go to https://infura.io/dashboard
+  2. Set up a free account
+  3. Make a new project
+
+      Your key should use "JSON-RPC over HTTPS" and look like `https://mainnet.infura.io/v3/<your project id>`
```
Comment (0xdef1cafe) [inline README.md:105]: this is duplicated from above, we should only include in one place

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
left some suggestions seth :)

File: README.md (hunk 5 above)
Comment (reidrankin) [reply README.md:105]: Dammit, flipping automerge decided to leave the duplicate stuff.

File: README.md [hunk 6]
```diff
@@ -114,3 +138,94 @@ From your projects folder:
...
+   git rebase -i upstream/develop
+   # Update your fork with the resolutions
+   git push --force-with-lease
+   ```
+
+6. Create a pull request on Github. (You can do this at `https://github.com/<username>/<fork name>/pull/new/<branch name>`.)
+
+   Ensure you've followed the guidelines in [CONTRIBUTING.md](https://github.com/shapeshift/web/blob/main/CONTRIBUTING.md); in particular, make sure that the title of your PR conforms to the Conventional Commits format.
+
+7. Post a link to your new pull request in `#engineering-prs` in the [Discord](https://discord.gg/shapeshift)
```
Comment (0xdef1cafe) [inline README.md:199]: this is great and belongs in this repo

File: README.md [hunk 7]
```diff
@@ -32,33 +32,57 @@ ShapeShift's OSS 2nd generation Web application. (Under Development)
...
+  Needed for certain Defi integrations such as Yearn; the app will malfunction when connecting a wallet without it.
+
+  Any Ethereum node should do, but you can get your own node URL for testing by doing the following:
+
+  1. Go to https://infura.io/dashboard
+  2. Set up a free account
+  3. Make a new project
+
+      Your key should use "JSON-RPC over HTTPS" and look like this: `https://mainnet.infura.io/v3/<your project id>`
+
```
Comment (0xdef1cafe) [inline README.md:85]: this is great and belongs in this repo

File: README.md [hunk 8]
```diff
@@ -32,33 +32,78 @@ ShapeShift's OSS 2nd generation Web application. (Under Development)
...
+Allows you to connect a Portis wallet. Without this the program will hang after choosing Portis and 
+clicking the "Pair" button. To get it, make a post asking for it in the Discord.
+
+REACT_APP_ETHEREUM_NODE_URL <br>
+Program crashes when connecting wallet without this. Obtain by doing the following: 
+1. Go to https://infura.io/dashboard
+2. Set up a free account
+3. Make a new project 
+<br>Your key should use "JSON-RPC over HTTPS" and look like this: 
+https://mainnet.infura.io/v3/YOUR-PROJECT-ID
```
Comment (reidrankin) [inline README.md:52]: remove accidental dupe of .env section below

```suggestion
```

File: README.md (hunk 4 above)
Comment (reidrankin) [reply README.md]: I'd support factoring this out into the docs repo but IMO that's better considered as a separate task

File: README.md (hunk 5 above)
Comment (reidrankin) [reply README.md]: This is the copy that should stay, https://github.com/shapeshift/web/pull/562#discussion_r774692867 should nuke the nukable one

File: README.md [hunk 9]
```diff
@@ -104,13 +128,21 @@ Runs the component documentation.<br /> Open
 
 ### Linking
 
-If you are working with any @shapeshiftoss dependencies use `yarn link`
-These next steps will show you how to link to the `chain-adapters` repo
-From your projects folder:
+The lib repository contains packages used in multiple ShapeShift repositories, and is often needed to run projects locally. 
+To use the latest version of lib when running the web app, it is necessary to use yarn link so that you can use local
+changes to lib in your local web repo.
```
Comment (0xdef1cafe) [inline README.md:133]: If you're developing locally in this `web` repository, and need to make changes affecting packages in `lib` or `unchained` (backend), use the following steps to link packages locally for developing. If your changes only touch `web` these steps are not necessary.

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Review by reidrankin — APPROVED** [review]

**Pull request merged**